resources/assets/lib/beatmap-discussions/editor-toolbar.tsx

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import * as _ from 'lodash';
+import { Portal } from 'portal';
+import * as React from 'react';
+import { Editor, Range } from 'slate';
+import { ReactEditor } from 'slate-react';
+import { isFormatActive, toggleFormat } from './editor-helpers';
+import { SlateContext } from './slate-context';
+
+const bn = 'beatmap-discussion-editor-toolbar';
+
+export class EditorToolbar extends React.Component {
+  static contextType = SlateContext;
+  ref: React.RefObject<HTMLDivElement>;
+  scrollContainer: HTMLElement | undefined;
+  private scrollTimer: number | undefined;
+  private readonly throttledUpdate: (() => null) & _.Cancelable; // TODO: is there a better way of typing this?
```

Comment:
Just assign the value here (`private readonly throttledUpdate = _.throttle(...)`)

Can also be done for the `ref` above I believe.